chord_metadata_service/chord/serializers.py

```diff
@@ -177,12 +199,29 @@ class ProjectSerializer(serializers.ModelSerializer):
     datasets = DatasetSerializer(read_only=True, many=True, exclude_when_nested=["project"])
     project_schemas = ProjectJsonSchemaSerializer(read_only=True, many=True)
 
+    counts = serializers.SerializerMethodField()
+
     # noinspection PyMethodMayBeStatic
     def validate_title(self, value):
         if len(value.strip()) < 3:
             raise serializers.ValidationError("Name must be at least 3 characters")
         return value.strip()
 
+    def get_counts(self, obj):
+        request = self.context.get("request")
+
+        # Only compute counts for read operations (GET), not write operations (POST/PUT/PATCH)
+        if not request or request.method not in ("GET", "HEAD", "OPTIONS"):
+            logger.debug(
+                "Skipping counts computation for non-GET request",
+                project_id=str(obj.identifier),
+                method=request.method if request else None,
+            )
+            return {}
+
+        scope = ValidatedDiscoveryScope(obj, None)
+        return get_censored_counts_for_serializer(request, scope)
```

Comment:
bind request method and `scope_repr` as described above to a new logger object, and then pass it here - that way any logs emitted in `get_censored_counts_for_serializer` contain all structured logging info (which is one of the cool parts of structured logging - the ability to bind more and more structured data as we drill the logger through functions for better debugging)